### PR TITLE
add probe X- and Y-offset to bed-leveling settings

### PR DESCRIPTION
**Pull request by salfter**
Most bed-leveling probes are offset from the nozzle.  For instance, when I center the nozzle on my printer at (110,110), the probe is at (84,70).  When probing the bed, this offset should be taken into account to get accurate x_adjust and y_adjust values.

### CODE REVIEW

**Comment by mamoit** [comment]
I like the idea behind I very much since I too have an offset on my probes

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ out
 *.pyc
 .config
 .config.old
+*~
```
Comment (mamoit) [inline .gitignore]: What is the purpose of this addition?

**Comment by mamoit** [comment]
Shouldn't the offsets be defined in the probe section?

**Comment by salfter** [comment]
I tried putting the settings in the probe section, but on a quick look through the code, I didn't see how to access the probe settings from within the bed-leveling code.

File: .gitignore (hunk 1 above)
Comment (salfter) [reply .gitignore]: It ignores the previous-version files saved by some editors.  (joe, for instance, creates foo~ when foo is saved.)  It makes for cleaner output when you're creating a commit.

Comment (mamoit) [reply .gitignore]: That is not on topic I think. But I guess it's up to the owners of the repo.

Comment (oderwat) [reply .gitignore]: You should just make that a global ignore on your system!

**Comment by salfter** [comment]
I have no idea what this "all checks have failed" thing is supposed to be about (clicking on "Details" brings up an empty page), but it might have something to do with my having committed my changes before pulling in updates from here.  I'm deleting and regenerating my fork and will put in another pull request from that.

**Pull request closed without merging**